subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/storage/SingleElementProcessingBucketsStorage.kt

```diff
@@ -0,0 +1,22 @@
+package com.avito.emcee.worker.internal.storage
+
+import com.avito.emcee.queue.Bucket
+
+internal class SingleElementProcessingBucketsStorage : ProcessingBucketsStorage {
+
+    private var bucket: Bucket? = null
+
+    override fun add(bucket: Bucket) {
+        require(this.bucket == null) { "There is a bucket with id=${bucket.bucketId} already" }
+        this.bucket = bucket
+    }
+
+    override fun remove(bucket: Bucket) {
+        require(this.bucket == bucket) { "There is no bucket with id=${bucket.bucketId}" }
+        this.bucket = null
+    }
+
```

Comment:
Shouldn't it be thread safe?

Reply:
Definitely yes, it should be. I'll make it thread-safe in the next PR.